pallets/avn/src/tests/tests.rs

```diff
@@ -1,64 +1,91 @@
+        let eth_index_before: u8 = AVN::get_primary_validator().0;
+
+        for _ in &validators {
+            AVN::calculate_primary_validator(OperationType::Ethereum).unwrap();
+        }
+
+        let eth_index_after: u8 = AVN::get_primary_validator().0;
+        assert_eq!(eth_index_before, eth_index_after);
     });
 }
 
 #[test]
-fn test_is_primary_blocknumber_3() {
+fn is_primary_function_correctly_returns_the_current_primary_validator_for_ethereum_operations() {
     let mut ext = ExtBuilder::build_default().with_validators().as_externality();
     ext.execute_with(|| {
-        let block_number = 3;
-        let expected_primary = 1;
-        let result = AVN::is_primary(block_number, &expected_primary);
+        let expected_primary = 0;
+        let result = AVN::is_primary(OperationType::Ethereum, &expected_primary);
+        assert!(result.is_ok(), "Getting primary validator failed");
+
+        AVN::calculate_primary_validator(OperationType::Ethereum);
+
+        let next_expected_primary = expected_primary + 1;
+        let result = AVN::is_primary(OperationType::Ethereum, &next_expected_primary);
         assert!(result.is_ok(), "Getting primary validator failed");
-        assert_eq!(result.unwrap(), true);
     });
 }
 
 #[test]
-fn test_is_primary_blocknumber_100() {
+fn next_validator_is_incremented_correctly_for_avn_operations() {
     let mut ext = ExtBuilder::build_default().with_validators().as_externality();
     ext.execute_with(|| {
-        let block_number = 100;
-        let expected_primary = 2;
-        let result = AVN::is_primary(block_number, &expected_primary);
-        assert!(result.is_ok(), "Getting primary validator failed");
-        assert_eq!(result.unwrap(), true);
+        let avn_index_before: u8 = AVN::get_primary_validator().1;
+        AVN::calculate_primary_validator(OperationType::Avn);
+        let avn_index_after: u8 = AVN::get_primary_validator().1;
+        assert_eq!(avn_index_after, avn_index_before + 1);
     });
 }
 
 #[test]
-fn is_primary_fails_with_no_validators() {
-    let mut ext = ExtBuilder::build_default().as_externality();
+fn the_first_validator_is_picked_again_after_the_last_validator_for_avn_operations() {
+    let mut ext = ExtBuilder::build_default().with_validators().as_externality();
     ext.execute_with(|| {
-        let block_number = 1;
-        let result = AVN::is_primary(block_number, &1);
-        assert!(result.is_err(), "Getting primary validator should have failed");
+        let validators = AVN::validators();
+        let eth_index_before: u8 = AVN::get_primary_validator().1;
+
+        for _ in &validators {
+            AVN::calculate_primary_validator(OperationType::Avn).unwrap();
+        }
+
+        let eth_index_after: u8 = AVN::get_primary_validator().1;
+        assert_eq!(eth_index_before, eth_index_after);
     });
 }
 
+#[test]
+fn is_primary_function_correctly_returns_the_current_primary_validator_for_avn_operations() {
```

Comment:
I don't understand the purpose of this test. What does the `is_primary_validator` do?
Shouldn't this function be returning true or false, and the tests check for that?
Also, `is_primary_validator` should **not** move the value of the next primary validator, since it should be a simple check, and not an action.
It would be valuable to have that test.